interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1029 @@
+
+    def build_text(self):
+        text = []
+        geometry = {"atom": "0", "linear": "1", "nonlinear": "2"}
+        for species in self.solution_species:
+            # each line contains the species name, integer representing
+            # geometry, Lennard-Jones potential well depth in K,
+            # Lennard-Jones collision diameter in angstroms,
+            # dipole moment in Debye,
+            # polarizability in cubic angstroms, and
+            # rotational relaxation collision number at 298 K.
+            line = []
+            line.append(f"{species.name:<16}")
+            line.append(f"{geometry[species.transport.geometry]:>4}")
+            line.append(f"{(species.transport.well_depth / ct.boltzmann):>10.3f}")
+            line.append(f"{(species.transport.diameter * 1e10):>10.3f}")
+            line.append(f"{(species.transport.dipole / DEBYE_CONVERSION):>10.3f}")
+            line.append(f"{(species.transport.polarizability * 1e30):>10.3f}")
+            line.append(f"{species.transport.rotational_relaxation:>10.3f} ! ")
+            if 'note' in species.input_data['transport']:
+                line.append(f"{species.input_data['transport']['note']}")
+
+            text.append("".join(line))
+        
+        return "\n".join(text)
+
+
+def convert(solution, mech_path=None, thermo_path=None, tran_path=None,
+            reorder_reactions=True, overwrite=False):
+    """
+    Writes Cantera solution object to Chemkin-format file(s).
+
+    :param solution:
+        Either the `solution` object being converted or the path leading to it
+    :param mech_path:
+        The Optional path to the output mechanism file, if this is unspecified it will
+        go to the cwd under the name in the `solution` object. 
+        It can be a file name with or without the extension, directory, or a full path
+    :param thermo_path:
+        The Optional path to the output thermo file, if this is unspecified it will
+        go to the cwd under the name in the `solution` object. 
+        It can be a file name with or without the extension, directory, or a full path
+    :param tran_path:
+        The Optional path to the output transport file, if this is unspecified it will
+        go to the cwd under the name in the `solution` object. 
+        It can be a file name with or without the extension, directory, or a full path
+    :param reorder_reactions
+        Boolean flag to reorder reactions based upon molecular weight
+    :param overwrite
+        Boolean flag to allow files to be overwritten or not
+    """
+    if not isinstance(solution, ct.Solution):
+        if not isinstance(solution, PurePath):
+            try:  # try to turn full path into pathlib Path
+                solution = Path(solution)
+            except:
+                raise ValueError(f"Invalid path: {str(solution)}")
+
+            if len(solution.parts) == 1:
+                solution = Path.cwd() / solution
+
```

Comment:
This is mostly because I find it convenient. It gives me the flexibility to provide either a relative or absolute path to a directory, file with or without an extension, or a file name with or without an extension. For people using this as a standalone script I send a script telling them where the files are written to (just in case). People using it as a script won't get that message because I don't want to force print statements. I suppose I could do a verbose option or just do the print statement and force them to intercept the console print statement if they don't want the text.

Reply:
@tsikes what I'm saying is that it's all redundant. As far as I can tell, you can get the behavior you want even without this code here, since the behavior you want is the default.